Github Actions failing for mypy on specific types libraries

Github Actions failing for mypy. Needs types libraries installed.

```
=================================== FAILURES ===================================
_________________________________ test session _________________________________
mypy exited with status 1.
____________________________ basin3d/core/access.py ____________________________
32: error: Library stubs not installed for "requests" (or incompatible with Python 3.7)
32: note: Hint: "python3 -m pip install types-requests"
32: note: (or run "mypy --install-types" to install all missing stub packages)
32: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
__________________________ basin3d/core/connection.py __________________________
17: error: Library stubs not installed for "requests" (or incompatible with Python 3.7)
18: error: Library stubs not installed for "yaml" (or incompatible with Python 3.7)
18: note: Hint: "python3 -m pip install types-PyYAML"
___________________________ basin3d/plugins/usgs.py ____________________________
52: error: Library stubs not installed for "requests" (or incompatible with Python 3.7)
________________________ tests/test_core_connection.py _________________________
4: error: Library stubs not installed for "requests" (or incompatible with Python 3.7)
```

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -22,7 +22,7 @@ jobs:
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
-        pip install flake8 pytest pytest-flake8 pytest-mypy pytest-cov $(cat docs/requirements.txt)
+        pip install flake8 pytest pytest-flake8 pytest-mypy pytest-cov types-PyYAML types-chardet types-cryptography types-requests $(cat docs/requirements.txt)
         if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
         # Output dependency versions
         pip freeze
@@ -32,12 +32,16 @@ jobs:
         flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
         # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
         flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+    - name: Static type checking with mypy
+      run: |
+        mypy -p basin3d  --python-version ${{ matrix.python-version }} > mypy.out
+        cat mypy.out
     - name: Make artifact directory
       run: |
         mkdir -p ${{ github.workspace }}/artifacts
     - name: Test with pytest
       run: |
-        PYTHONPATH=. pytest -v -m "not flake8" --cov=basin3d --cov-report html:${{ github.workspace }}/artifacts
+        PYTHONPATH=. pytest -v -m "not flake8 and not mypy" --cov=basin3d --cov-report term --cov-report html:${{ github.workspace }}/artifacts
     - name: Documentation and Doctests
       run: |
         cd docs

